code/src/sixsq/nuvla/server/resources/spec/voucher.cljc

```diff
@@ -0,0 +1,189 @@
+(ns sixsq.nuvla.server.resources.spec.voucher
+  (:require
+    [clojure.spec.alpha :as s]
+    [sixsq.nuvla.server.resources.spec.common :as cimi-common]
+    [sixsq.nuvla.server.resources.spec.core :as cimi-core]
+    [sixsq.nuvla.server.util.spec :as su]
+    [spec-tools.core :as st]))
+
+
+(s/def ::owner ::cimi-common/resource-link)
+(s/def ::user ::cimi-common/resource-link)
```

Comment:
If these are links to the user resources, it would be better to reuse the `::id` definition in `user.cljc`.  You should also provide metadata for these fields.

Reply:
changed